crmsh/corosync.py

```diff
@@ -669,215 +649,370 @@ def remove_qdevice_db(self):
         parallax.parallax_call(node_list, cmd, self.askpass)
 
 
-def corosync_tokenizer(stream):
-    """Parses the corosync config file into a token stream"""
-    section_re = re.compile(r'(\w+)\s*{')
-    value_re = re.compile(r'(\w+):\s*([\S ]+)')
-    path = []
-    while stream:
-        stream = stream.lstrip()
-        if not stream:
-            break
-        if stream[0] == '#':
-            end = stream.find('\n')
-            t = Token(_tCOMMENT, [], stream[:end])
-            stream = stream[end:]
-            yield t
-            continue
-        if stream[0] == '}':
-            t = Token(_tEND, [])
-            stream = stream[1:]
-            yield t
-            path = path[:-1]
-            continue
-        m = section_re.match(stream)
-        if m:
-            path.append(m.group(1))
-            t = Token(_tBEGIN, path, m.group(1))
-            stream = stream[m.end():]
-            yield t
-            continue
-        m = value_re.match(stream)
-        if m:
-            t = Token(_tVALUE, path + [m.group(1)], m.group(1), m.group(2))
-            stream = stream[m.end():]
-            yield t
-            continue
-        raise ValueError("Parse error at [..%s..]" % (stream[:16]))
+class Parser(object):
+    """
+    Class to parse config file which syntax is similar with corosync.conf
+    Parse corosync.conf by default
+    """
+    INDENT = 4
 
+    def __init__(self, config_file=None):
+        """
+        Init function
+        """
+        self._config_file = config_file if config_file else conf()
```

Comment:
The `conf()` function (defined `crmsh/corosync.py`) uses the Corosync config file as string. However, that string is hardcoded in other places:

```
$ grep -n /etc/corosync/corosync.conf crmsh/*.py 
crmsh/bootstrap.py:1249:include /etc/corosync/corosync.conf;
crmsh/corosync.py:22:    return os.getenv('COROSYNC_MAIN_CONFIG_FILE', '/etc/corosync/corosync.conf')
crmsh/utils.py:1112:    elif os.path.exists("/etc/corosync/corosync.conf") or is_program('corosync-cfgtool'):
crmsh/utils.py:1675:        cfg = os.getenv('COROSYNC_MAIN_CONFIG_FILE', '/etc/corosync/corosync.conf')
crmsh/utils.py:1694:        cfg = os.getenv('COROSYNC_MAIN_CONFIG_FILE', '/etc/corosync/corosync.conf')
```

I know, it's not directly related to this PR. Maybe create a constant and import that.

Reply:
@tomschr Thanks for pointing this
Already added to my backlog

Reply:
This line could be improve with:
```
self._config_file = config_file or conf()
```